WooCommerce/Classes/ViewRelated/Products/Edit Product/EditCategories/ProductCategoryListViewModel.swift

```diff
@@ -0,0 +1,74 @@
+import Foundation
+import Yosemite
+
+final class ProductCategoryListViewModel {
+
+    private let product: Product
+
+    private lazy var categoriesResultController: ResultsController<StorageProductCategory> = {
+        let storageManager = ServiceLocator.storageManager
+        let predicate = NSPredicate(format: "siteID = %ld", self.product.siteID)
+        let descriptor = NSSortDescriptor(keyPath: \StorageProductCategory.name, ascending: true)
+        return ResultsController<StorageProductCategory>(storageManager: storageManager, matching: predicate, sortedBy: [descriptor])
+    }()
+
+    init(product: Product) {
+        self.product = product
+        performInitialFetch()
```

Comment:
True, I had the `performInitialFetch()` in the `observeCategories` method but that also seemed a bit miss-leading  so I ended up moving it up. I'll change it for `configureViewModel` since it's more aligned with all others VCs

Reply:
Fixed https://github.com/woocommerce/woocommerce-ios/pull/2112/commits/5f85e8c90c8cbc6d9b44d631a4629bd8c239702f